internal/metrics/types.go

```diff
@@ -158,13 +160,22 @@ type MetricsSourceProvider interface {
 	Timeout() time.Duration
 }
 
+type Event struct {
```

Comment:
nitpick: This doesn't belong under `internal/metrics`. Should be okay to create a `internal/events/types.go` and define an `EventSink` interface with the `Name()`, `Stop()` and `ExportEvents(...)` methods. The Wavefront sink can implement both interfaces and used accordingly.

Reply:
I don't see why we need all those interfaces... we just have one sink type, and this is not a API...